src/gui/processing/qgsprocessingmeshdatasetwidget.cpp

```diff
@@ -0,0 +1,568 @@
+      {
+        setMeshLayerWrapperValue( layerParameterWrapper );
+        setDatasetGroupIndexesWrapperValue( datasetGroupsParameterWrapper );
+      } );
+
+      break;
+    }
+
+    case QgsProcessingGui::Modeler:
+      break;
+  }
+}
+
+void QgsProcessingMeshDatasetTimeWidgetWrapper::setMeshLayerWrapperValue( const QgsAbstractProcessingParameterWidgetWrapper *wrapper )
+{
+  if ( !mWidget )
+    return;
+
+  // evaluate value to layer
+  QgsProcessingContext *context = nullptr;
+  std::unique_ptr< QgsProcessingContext > tmpContext;
+  if ( mProcessingContextGenerator )
+    context = mProcessingContextGenerator->processingContext();
+
+  if ( !context )
+  {
+    tmpContext = qgis::make_unique< QgsProcessingContext >();
+    context = tmpContext.get();
+  }
+
+  QgsMeshLayer *meshLayer = QgsProcessingParameters::parameterAsMeshLayer( wrapper->parameterDefinition(), wrapper->parameterValue(), *context );
+
+  if ( mWidget )
+    mWidget->setMeshLayer( meshLayer );
+
+}
+
+void QgsProcessingMeshDatasetTimeWidgetWrapper::setDatasetGroupIndexesWrapperValue( const QgsAbstractProcessingParameterWidgetWrapper *wrapper )
+{
+  if ( !mWidget )
+    return;
+
+  QVariant datasetGroupsVariant = wrapper->parameterValue();
+
+  if ( !datasetGroupsVariant.isValid() || datasetGroupsVariant.type() != QVariant::List )
+    mWidget->setDatasetGroupIndexes( QList<int>() );
+
+  QVariantList datasetGroupsListVariant = datasetGroupsVariant.toList();
+
+  QList<int> datasetGroupsIndexes;
+  for ( const QVariant &variantIndex : datasetGroupsListVariant )
+    datasetGroupsIndexes << variantIndex.toInt();
+
+  mWidget->setDatasetGroupIndexes( datasetGroupsIndexes );
+
+}
+
+QWidget *QgsProcessingMeshDatasetTimeWidgetWrapper::createWidget()
+{
+  mWidget = new QgsProcessingMeshDatasetTimeWidget( nullptr, static_cast<const QgsProcessingParameterMeshDatasetTime *>( parameterDefinition() ), widgetContext() );
+
+  QgsMapCanvas *canvas = widgetContext().mapCanvas();
+  if ( canvas )
+  {
+    connect( canvas, &QgsMapCanvas::temporalRangeChanged, mWidget, &QgsProcessingMeshDatasetTimeWidget::buildValue );
+  }
+  connect( mWidget, &QgsProcessingMeshDatasetTimeWidget::changed, this, [ = ]
+  {
+    emit widgetValueHasChanged( this );
+  } );
+
+  return mWidget;
+}
+
+void QgsProcessingMeshDatasetTimeWidgetWrapper::setWidgetValue( const QVariant &value, QgsProcessingContext &context )
+{
+  if ( mWidget )
+    mWidget->setValue( value );
+}
+
+QVariant QgsProcessingMeshDatasetTimeWidgetWrapper::widgetValue() const
+{
+  if ( mWidget )
+    return mWidget->value();
+  return QVariant();
+}
+
+QgsProcessingMeshDatasetTimeWidget::QgsProcessingMeshDatasetTimeWidget( QWidget *parent,
+    const QgsProcessingParameterMeshDatasetTime *param,
+    const QgsProcessingParameterWidgetContext &context ):
+  QWidget( parent ),
+  mParam( param )
+{
+  setupUi( this );
+
+  dateTimeEdit->setDisplayFormat( "yyyy-MM-dd HH:mm:ss" );
```

Comment:
Set the range for dateTimeEdit to the maximum possible -- e.g. see QgsTemporalControllerWidget

Reply:
hmmm, it is set to max in the ui file